### PR TITLE
STY: de-privatize names imported across modules

### PR DESCRIPTION
**Pull request by jbrockmendel**
getting close to the end of this and enabling a code_check

### CODE REVIEW

File: asv_bench/benchmarks/sparse.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def setup(self):
         s.index = MultiIndex.from_product([range(10)] * 4)
         self.ss = s.astype("Sparse")
 
-    def time_sparse_series_to_coo(self):
+    def timesparse_series_to_coo(self):
```
Comment (jreback) [inline asv_bench/benchmarks/sparse.py:74]: leave as is

File: pandas/core/index.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
     ensure_index_from_sequences,
     get_objs_combined_axis,
 )
-from pandas.core.indexes.multi import _sparsify  # noqa:F401
+from pandas.core.indexes.multi import sparsify_labels as _sparsify  # noqa:F401
```
Comment (jreback) [inline pandas/core/index.py:22]: can you make the edit inside this

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, 2 comments

File: pandas/core/index.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/index.py:22]: not sure what you're asking for here

File: asv_bench/benchmarks/sparse.py (hunk 1 above)
Comment (jbrockmendel) [reply asv_bench/benchmarks/sparse.py:74]: woops, will update

File: pandas/core/index.py (hunk 2 above)
Comment (jreback) [reply pandas/core/index.py:22]: don't use the as import
just update to use sparsify_labels instead of _sparsify

**Comment by jbrockmendel** [comment]
updated+green

**Review by jreback — APPROVED** [review]

File: pandas/core/index.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
     ensure_index_from_sequences,
     get_objs_combined_axis,
 )
-from pandas.core.indexes.multi import _sparsify  # noqa:F401
+from pandas.core.indexes.multi import sparsify_labels  # noqa:F401
```
Comment (jreback) [inline pandas/core/index.py:22]: actually i think you can just remove this line entirely (its not used inside)

Comment (jbrockmendel) [reply pandas/core/index.py:22]: i mean, nothing is used in this file

Comment (jreback) [reply pandas/core/index.py:22]: ok, then wonder if we need this *at* all here (and maybe even re-evaluate *all* of these exports); but let's do later.

Comment (jbrockmendel) [reply pandas/core/index.py:22]: i think the idea is to remove the file in 2.0

**Pull request merged**